src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,563 @@
+     *     <td>"{0} and {1}"</td>
+     * <tr><th scope="row" style="text-align:left">3</th>
+     *     <td>""</td>
+     * </tbody>
+     * </table>
+     * Here are the resulting formatted strings with the above pattern array.
+     * <table class="striped">
+     * <caption style="display:none">Formatting examples</caption>
+     * <thead>
+     * <tr><th scope="col">Input String Array</th>
+     *     <th scope="col">Formatted String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz", "Qux"]</th>
+     *     <td>"Foo, Bar, Baz, and Qux"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz"]</th>
+     *     <td>"Foo, Bar, and Baz"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar"]</th>
+     *     <td>"Foo and Bar"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo"]</th>
+     *     <td>"Foo"</td>
+     * </tbody>
+     * </table>
+     *
+     * @param patterns array of patterns, not null
+     * @throws IllegalArgumentException if the length {@code patterns} array is less than 5, or
+     *          any of {@code start}, {@code middle}, {@code end} patterns cannot be parsed.
+     * @throws NullPointerException if {@code patterns} is null.
+     */
+    public static ListFormat getInstance(String[] patterns) {
+        Objects.requireNonNull(patterns);
+        if (patterns.length != PATTERN_ARRAY_LENGTH) {
+            throw new IllegalArgumentException("Pattern array length should be " + PATTERN_ARRAY_LENGTH);
+        }
+        return new ListFormat(Locale.ROOT, Arrays.copyOf(patterns, PATTERN_ARRAY_LENGTH));
+    }
+
+    /**
+     * {@return the string that consists of the input strings, concatenated with the
+     * patterns of this {@code ListFormat}}
+     * @param input The list of input strings to format. There should at least
+     *              one String element in this list, otherwise an {@code IllegalArgumentException}
+     *              is thrown.
+     * @throws IllegalArgumentException if the length of {@code input} is zero.
+     * @throws NullPointerException if {@code input} is null.
+     */
+    public String format(List<String> input) {
+        Objects.requireNonNull(input);
+        return format(input, new StringBuffer(),
+                DontCareFieldPosition.INSTANCE).toString();
+    }
+
+    /**
+     * Formats an object and appends the resulting text to a given string
+     * buffer. The object should either be a List or an array of Objects.
+     * If the {@code pos} argument identifies a field used by the format,
+     * then its indices are set to the beginning and end of the first such
+     * field encountered.
+     *
+     * @param obj    The object to format. Must be a List or an array
+     *               of Object.
+     * @param toAppendTo    where the text is to be appended
+     * @param pos    A {@code FieldPosition} identifying a field
```

Comment:
It should be clear that `pos` is an output parameter describing where in the StringBuffer the list was written.

Reply:
Will clarify it in the next iteration.

Reply:
In fact, ListFormat should ignore it, as it does not use the subformats underneath.